scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -71,3 +81,62 @@ def __len__(self) -> int:
 
     def __iter__(self):
         return iter({self._name: value} for value in self._values)
+
+
+class AttackPointIteratorInternalBalancedGenerator(
+        AttackPointIteratorInternalBase):
+    """
+    Attack point iterator class that iterates over the balanced generator.
+    """
+
+    def __init__(self,
+                 operation: str,
+                 name: str,
+                 length: int,
+                 bunches: int = 1,
+                 elements: int = 256) -> None:
+        """Initialize the balanced kwargs to iterate."""
+        self._operation = operation
+        self._name = name
+        self._length = length
+        self._bunches = bunches
+        self._elements = elements
+        self._len = self._bunches * self._elements
+
+    def __len__(self) -> int:
+        return self._len
+
+    def __iter__(self):
+        return iter({self._name: value}
+                    for value in balanced_generator(length=self._length,
+                                                    bunches=self._bunches,
+                                                    elements=self._elements))
+
+
+class AttackPointIteratorInternalUnrestrictedGenerator(
+        AttackPointIteratorInternalBase):
+    """
+    Attack point iterator class that iterates over the unrestricted generator.
+    """
+
+    def __init__(self,
+                 operation: str,
+                 name: str,
+                 length: int,
+                 elements: int = 256,
+                 bunches: int = 1) -> None:
+        """Initialize the unrestricted kwargs to iterate."""
+        self._operation = operation
+        self._name = name
+        self._length = length
+        self._elements = elements
+        self._bunches = bunches
+        self._len = self._bunches * self._elements
+
+    def __len__(self) -> int:
```

Comment:
I would add a unittest for this case as well (since there was a bug in balanced generator).